lib/js/components/Buttons/IconButton/IconButton.vue

```diff
@@ -50,42 +57,83 @@
 @import '../../../../styles/settings/spacings';
 @import '../../../../styles/settings/typography';
 
-@mixin iconButtonColor($color, $hover-color: '') {
-	@if $hover-color == '' {
-		//$hover-color: mix($color, $color-firefly-black, (1 - $button-hover-alpha) * 100%);
-	}
-
-	color: $color;
-
-	&:hover,
-	&:active {
-		color: $hover-color;
-	}
-}
-
 .a-iconButton {
 	$self: &;
 
-	@each $color-name, $color-map in $theme-calculated-colors {
+	@each $color-name, $color-map in $icon-button-colors {
 		&.-color-#{$color-name} {
-			@include iconButtonColor(map-get($color-map, 'color'), map-get($color-map, 'hovered'));
-		}
-	}
+			#{$self}__label {
+				color: map-get($color-map, 'outlined', 'color');
+			}
 
-	@each $color-name, $color in $regular-colors {
-		&.-color-#{$color-name} {
-			@include iconButtonColor($color);
+			&:hover,
+			&.-hovered {
+				#{$self}__button {
+					background-color: map-get($color-map, 'filled', 'background-hovered');
+					color: map-get($color-map, 'icon');
+
+					&.-outlined {
+						background-color: map-get($color-map, 'outlined', 'background-hovered');
+						color: map-get($color-map, 'inverted', 'icon');
+					}
+				}
+				#{$self}__label {
+					color: map-get($color-map, 'filled', 'color-hovered');
+				}
+			}
+
+			&:focus,
+			&.-focused {
+				#{$self}__button {
+					background-color: map-get($color-map, 'filled', 'background-focused');
+					color: map-get($color-map, 'filled', 'icon');
+
+					&.-outlined {
+						background-color: map-get($color-map, 'outlined', 'background-focused');
+						color: map-get($color-map, 'inverted', 'icon');
+					}
+				}
+				#{$self}__label {
+					color: map-get($color-map, 'filled', 'color-focused');
+				}
+			}
+
+			&.-disabled {
+				#{$self}__button {
+					background-color: map-get($color-map, 'filled', 'background-disabled');
+					color: map-get($color-map, 'icon');
+
+					&.-outlined {
+						background-color: map-get($color-map, 'outlined', 'background');
+						border-color: map-get($color-map, 'outlined', 'disabled', 'border');
+
+						.a-iconButton__icon {
+							color: map-get($color-map, 'outlined', 'disabled', 'icon');
+						}
+					}
+				}
+
+				#{$self}__label {
+					color: map-get($color-map, 'outlined', 'disabled', 'color');
+				}
+			}
 		}
 	}
 
 	align-items: center;
-	color: $color-primary;
+	color: $color-primary-text;
 	cursor: pointer;
 	display: inline-flex;
 	transition: color ease-in-out $default-transition-time;
 
+	&:disabled,
+	&.-disabled {
+		cursor: not-allowed;
```

Comment:
it won't work, because we disabled `pointer-events`

Reply:
It's not my idea, but right this rule do nothing and it's redundant. I will remove it.